src/main/java/com/spanish_inquisition/battleship/server/game_states/PlacingShipsState.java

```diff
@@ -10,29 +13,25 @@
 import static com.spanish_inquisition.battleship.server.BattleshipServer.SERVER_ID;
 
 public class PlacingShipsState extends GameState {
-    private List<Player> players;
-    private MessageBus requestBus;
-
-    public PlacingShipsState(List<Player> players, MessageBus requestBus) {
-        this.players = players;
-        this.requestBus = requestBus;
+    public PlacingShipsState(Players players, MessageBus requestBus) {
+        super(players, requestBus);
     }
 
     @Override
     public GameState transform() {
-        List<Player> notReadyPlayers = new LinkedList<>(players);
-        while(!notReadyPlayers.isEmpty()) {
-            for(Player player : notReadyPlayers) {
-                if(requestBus.haveMessageFromSender(player.getPlayerId())) {
+        List<Player> notReadyPlayers = new LinkedList<>(players.getBothPlayers());
+        while (!notReadyPlayers.isEmpty()) {
+            for (Player player : notReadyPlayers) {
+                if (requestBus.haveMessageFromSender(player.getPlayerId())) {
                     resolvePlayerFleetValidation(player, notReadyPlayers);
                 }
             }
         }
-        return new EndState();
+        return new PlayerActionState(players, requestBus);
     }
 
     private void resolvePlayerFleetValidation(Player player, List<Player> notReadyPlayers) {
-        if(setFleetIfIsValid(player)) {
+        if (setFleetIfIsValid(player)) {
```

Comment:
I would split it and actually send an already valid fleet here just to place it.